server/backend/api/app/schemas/fam_user_role_assignment_create.py

```diff
@@ -16,5 +29,39 @@ class FamUserRoleAssignmentCreateSchema(BaseModel):
         List[Annotated[str, StringConstraints(min_length=1, max_length=8)]], None
     ] = None
     requires_send_user_email: bool = False
+    expiry_date_date: Optional[str] = None  # e.g. '2025-12-31', BC timezone.
+    # Internal use only, not exposed in OpenAPI or request/response bodies
+    _expiry_date: Optional[datetime] = PrivateAttr(default=None)
+
+
+    @field_validator('expiry_date_date')
+    @classmethod
+    def validate_expiry_date_date(cls, v):
+        if v is None:
+            return v
+        if isinstance(v, str) and v.strip() == "":
+            return None
+        bc_tz = ZoneInfo(BC_TIMEZONE)
+        try:
+            dv = datetime.strptime(v, DATE_FORMAT_YYYY_MM_DD).date()
+        except Exception:
+            raise ValueError('expiry_date_date must be a valid YYYY-MM-DD string')
+        now_bc = datetime.now(bc_tz).date()
```

Comment:
I'm not sure I understand why this logic is necessary. If the user enters the expiry date, shouldn't that just be what we save. The conversion to BC timezone only makes sense if we are going to store the date & time, but I'm not sure we are doing that. And if we are storing the date & time, shouldn't we be setting the time component to midnight somewhere in here?